test/steps/image_registry.go

```diff
@@ -40,7 +40,7 @@ func (data *Data) localServiceBuiltByMavenWithProfileAndDeployedToRuntimeRegistr
 		return err
 	}
 
-	if err := framework.GetContainerEngine(data.Namespace).PullImage(imageTag).GetError(); err != nil {
+	if needToBuildImage(data.Namespace, imageTag) {
 		// Not found in registry, so we need to build and push the application
 		// Build the application
 		err = data.localServiceBuiltByMavenWithProfile(contextDir, profile)
```

Comment:
This might be unrelated to these changes, but I think the image built does not take into account the parameters: **build_image_version** and **build_image_name_suffix**. 

These parameters are used to take the docker image base from, but using the image cache this issue might be worse as we will be reusing an image with a base image, when I asked to use another base image. Probably, we could amend [this line](https://github.com/kiegroup/kogito-cloud-operator/blob/25122c0e2a0236357a8b6d8dd2f940db5465cf31/test/steps/image_registry.go#L109) to use the value from [GetBuildImage](https://github.com/kiegroup/kogito-cloud-operator/blob/25122c0e2a0236357a8b6d8dd2f940db5465cf31/test/framework/util.go#L222) instead of latest.

Reply:
good point, will apply

Reply:
Well, when I think about it, it may have more sense to create a separate `runtime_application_image_version` and `runtime_application_image_name_suffix` to keep consistency of the configuration (as we already have `runtime_application_image_registry` and `runtime_application_image_namespace`)